### DISCUSSION TITLE
Update the tips collection without updating the module

### DISCUSSION THREAD

**Discussion by joshooaj** [deadlydog/PowerShell.tiPS#59] [Ideas]
Hi Dan, nice demo at RTPSUG today!

You mentioned that you get new tips by installing the most recent version of the module, but I was thinking you could have the module attempt to pull the latest tips from [https://raw.githubusercontent.com/deadlydog/PowerShell.tiPS/main/src/tiPS/PowerShellTips.json](https://raw.githubusercontent.com/deadlydog/PowerShell.tiPS/main/src/tiPS/PowerShellTips.json) once in a while.

It could be done in the background (a runspace or job or something), and maybe check no more than once per day by default. That way you and others can submit tips over time and anyone who as the module installed will get the new tips without a whole module update.

**Comment by deadlydog** [comment]
Thanks for the feedback Josh! I hadn't considered pulling just the JSON file down directly from GitHub. I think I may have glossed over it during the RTPSUG lightning demo, but the module currently supports auto-updating itself using a background job.

From [the Quick Start](https://github.com/deadlydog/PowerShell.tiPS?tab=readme-ov-file#-quick-start):

```powershell
Set-TiPSConfiguration -AutomaticallyWritePowerShellTip Daily -AutomaticallyUpdateModule Weekly
```

The `-AutomaticallyUpdateModule Weekly` parameter tells the module to automatically update itself each week to ensure users always have the latest stable version with the new tips. You can also provide `Daily` to ensure the module updates even more often. There's a bit more info in [the Automatic Updates section](https://github.com/deadlydog/PowerShell.tiPS?tab=readme-ov-file#-automatic-updates).

Since the module is already able to update itself Daily/Weekly/Biweekly/Monthly, I'm not sure if we need the additional functionality of downloading and updating _just_ the JSON file. Do you see a reason for that which I'm overlooking? Or maybe it just wasn't clear that the module keeps itself up to date automatically without the user having to do anything?

Thanks!

**Comment by joshooaj** [comment]
Fair enough Dan! I did catch the bit about it self-updating the module and I have no other reason to suggest just updating the JSON file other than it "feels right" to pull the latest tips on their own as a smaller package when I imagine they would be updated a lot more frequently over time than the module itself.

You've already got a pretty slick self-updating process implemented though and I dig that it cleans up after itself by removing the old versions so you don't end up with a stack of old modules over time. Carry on friend 😎
